packages/space-opera/src/components/materials_panel/materials_panel.ts

```diff
@@ -295,358 +309,257 @@ export class MaterialPanel extends ConnectedLitElement {
-        getEditsMaterials(reduxStore.getState()),
-        getEditsTextures(reduxStore.getState()),
-        {id, uri}));
+    this.onTextureUpload(event.detail, this.getMaterial().emissiveTexture);
   }
 
   onEmissiveFactorChanged() {
-    if (this.selectedMaterialId === undefined) {
-      throw new Error('No material selected');
-    }
-
-    const id = this.selectedMaterialId;
-    const emissiveFactor = this.selectedEmissiveFactor;
-    reduxStore.dispatch(dispatchSetEmissiveFactor(
-        getEditsMaterials(reduxStore.getState()), {id, emissiveFactor}));
+    this.getMaterial().setEmissiveFactor(this.selectedEmissiveFactor);
+    reduxStore.dispatch(dispatchModelDirty());
   }
 
   onOcclusionTextureChange() {
-    if (this.selectedMaterialId === undefined) {
-      throw new Error('No material selected');
-    }
-
-    const id = this.selectedMaterialId;
-    const textureId = this.selectedOcclusionTextureId;
-    reduxStore.dispatch(dispatchOcclusionTexture(
-        getEditsMaterials(reduxStore.getState()), {id, textureId}));
+    this.onTextureChange(
+        this.selectedOcclusionTextureId, this.getMaterial().occlusionTexture);
   }
 
   onOcclusionTextureUpload(event: CustomEvent) {
-    if (this.selectedMaterialId === undefined) {
-      throw new Error('No material selected');
-    }
-
-    const id = this.selectedMaterialId;
-    const uri = event.detail;
-    reduxStore.dispatch(dispatchAddOcclusionTexture(
-        getEditsMaterials(reduxStore.getState()),
-        getEditsTextures(reduxStore.getState()),
-        {id, uri}));
+    this.onTextureUpload(event.detail, this.getMaterial().occlusionTexture);
   }
 
   onAlphaModeSelect() {
-    if (this.selectedMaterialId === undefined) {
-      throw new Error('No material selected');
-    }
-
     const selectedMode =
-        this.alphaModePicker?.selectedItem?.getAttribute('value');
-
-    if (!selectedMode) {
-      return;
-    }
-
-    reduxStore.dispatch(
-        dispatchSetAlphaMode(getEditsMaterials(reduxStore.getState()), {
-          id: this.selectedMaterialId,
-          alphaMode: selectedMode,
-        }));
+        ALPHA_BLEND_MODES[this.alphaModePicker.selectedIndex] as AlphaMode;
+    this.alphaCutoffContainer.style.display =
+        selectedMode === 'MASK' ? '' : 'none';
+    const material = this.getMaterial()
+    material.setAlphaMode(selectedMode);
+    this.alphaCutoffSlider.value = material.getAlphaCutoff();
+    reduxStore.dispatch(dispatchModelDirty());
   }
 
   onAlphaCutoffChange() {
-    if (this.selectedMaterialId === undefined) {
-      throw new Error('No material selected');
-    }
-
-    reduxStore.dispatch(
-        dispatchSetAlphaCutoff(getEditsMaterials(reduxStore.getState()), {
-          id: this.selectedMaterialId,
-          alphaCutoff: this.selectedAlphaCutoff,
-        }));
+    this.getMaterial().setAlphaCutoff(this.selectedAlphaCutoff);
+    reduxStore.dispatch(dispatchModelDirty());
   }
 
   revertMetallicRoughnessTexture() {
```

Comment:
All these revert functions look awful similar, seems like a good candidate for refactoring into one helper method with a parameter to indicating the texture to revert. Then we could delete a whole lot more code.

Reply:
I think I'm going to leave this for now in the interest of getting this merged. There's plenty more cleanup to do in space-opera when someone is excited about it.